### PR TITLE
[draft] Initial GW implementation

### PR DESCRIPTION
**Pull request by r57**
Adding Apollo Gateway with federated services and code generation for GQL and gRPC (proto files pending from Rudolf). This also slightly changes GQL schema.

### CODE REVIEW

**Comment by cyberduck0** [comment]
Everything looks good to me.
There is one thing which I will ask you to add tho.
We need a way to sort the ToDoItems, so if possible please add an `index` field.
So the UI can sort the items and change order as we will have drag & drop.

File: gateway/src/model/todo.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export interface Todo {
+    id: string;
+    title: string;
+    comment: string;
```
Comment (gemma-fevo) [inline gateway/src/model/todo.ts]: Doesn't this need `items` as well? Sorry if it's a stupid question 🙈

Comment (cyberduck0) [reply gateway/src/model/todo.ts]: The todo items are separated from the main object here.
Also, I think it would be better if it does...?

Comment (r57) [reply gateway/src/model/todo.ts]: They're separated inside of the app model, yes. I think it's better that way. They're glued together with GQL later.

File: gateway/src/model/todo.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+export interface TodoItem {
```
[the remark this answers is not in the document]
Comment (r57) [reply gateway/src/model/todo.ts]: I'll add in another PR shall I?

**Comment by r57** [comment]
@cyberduck0 I'd not make `items` part of the `todo` as user will rarely operate on all of them at one time so they'd be dragged back and fort for no reason.

File: gateway/src/model/todo.ts (hunk 1 above)
Comment (gemma-fevo) [reply gateway/src/model/todo.ts]: ok cool, sounds good :)

**Pull request merged**